src/core_read.cpp

```diff
@@ -88,18 +88,38 @@ CScript ParseScript(const std::string& s)
     return result;
 }
 
+// Check that all of the input and output scripts of a transaction contains valid opcodes
+bool CheckTxScriptsSanity(CMutableTransaction& tx)
+{
+    // Check input scripts
+    for(unsigned int i = 0; i < tx.vin.size(); i++) {
```

Comment:
You should probably skip this loop for coinbase transactions, as those don't have meaningful input scripts.

Is testing the inputs even useful, as the function of this test is to distinguish between transactions that may errorneously be decoded as 0-input transaction?

Nit: space after `for` (and elsewhere).